awx/main/dispatch/__init__.py

```diff
@@ -64,9 +64,9 @@ def events(self, select_timeout=5, yield_timeouts=False):
                 if yield_timeouts:
                     yield None
             else:
-                self.conn.poll()
-                while self.conn.notifies:
-                    yield self.conn.notifies.pop(0)
+                notification_generator = self.conn.notifies()
+                for notification in notification_generator:
```

Comment:
I was trying to read

https://www.psycopg.org/psycopg3/docs/advanced/async.html#asynchronous-notifications

But it looks like you're done it and gotten the key pattern change here. Little weird to save `notification_generator`, but this is all fine. This is a documented change from the library, and is cleaner than before.